clutel/clutel.go

```diff
@@ -32,3 +34,88 @@ func AddToOTELHTTPLabeler(
 		node.AddToOTELHTTPLabeler(labeler),
 	))
 }
+
+// GetSpan retrieves the current OpenTelemetry span from the context.
+func GetSpan(ctx context.Context) trace.Span {
+	return trace.SpanFromContext(ctx)
+}
+
+type spanBuilder struct {
+	kvs  map[string]any
+	opts []trace.SpanStartOption
+}
+
+// NewSpan produces a span builder that allows complete configuration and
+// attribution of the span before it gets started.
+func NewSpan() *spanBuilder {
+	return &spanBuilder{}
+}
+
+// WithAttrs adds attrs to the span.
+func (sb *spanBuilder) WithAttrs(kvs ...any) *spanBuilder {
+	if sb.kvs == nil {
+		sb.kvs = make(map[string]any)
+	}
+
+	sb.kvs = stringify.Normalize(kvs...)
+
+	return sb
+}
+
+// WithOpts configures the span with specific otel options.  Note that one
+// of the options can be `trace.WithAttributes()`; don't use that, use
+// spanBuilder.WithAttrs() instead.
+func (sb *spanBuilder) WithOpts(opts ...trace.SpanStartOption) *spanBuilder {
+	if sb.opts == nil {
+		sb.opts = make([]trace.SpanStartOption, 0, len(opts))
+	}
+
+	sb.opts = append(sb.opts, opts...)
+
+	return sb
+}
+
+// Start begins the span with the provided name and attaches it to the context.
+func (sb *spanBuilder) Start(
+	ctx context.Context,
+	name string,
+) context.Context {
+	nc := node.FromCtx(ctx)
+	if nc == nil {
+		return ctx
+	}
+
+	ctx, spanned := nc.AddSpan(ctx, name, sb.opts...)
```

Comment:
not directly related to this PR, but I think there's a bug in the logic for `AddValues`. I think it should be `if !cfg.doNotAddToSpan | ...` [here](https://github.com/alcionai/clues/blob/85d28fb51c603ecc3ea75184bcfdec6edb065836/internal/node/node.go#L145)

Reply:
Right you are!  Good catch.